src/testdir/test_cd.vim

```diff
@@ -252,4 +252,43 @@ func Test_getcwd_actual_dir()
   call chdir(startdir)
 endfunc
 
+func Test_cd_with_symlink()
```

Comment:
This only tests the new behaviour, but it should also test the current behaviour, no?